aws/ebs.go

```diff
@@ -47,16 +47,19 @@ func nukeAllEbsVolumes(session *session.Session, volumeIds []*string) error {
 
 		_, err := svc.DeleteVolume(params)
 		if err != nil {
-			// Ignore not found errors, some volumes are deleted along with EC2 Instances
-			if !strings.Contains(err.Error(), "InvalidVolume.NotFound") {
-				logging.Logger.Errorf("[Failed] %s", err)
-				return errors.WithStackTrace(err)
+			if awsErr, isAwsErr := err.(awserr.Error); isAwsErr && awsErr.Code() == "VolumeInUse" {
```

Comment:
This error message could mean other things, so I'm not sure assuming it's a protected instance is the right move here. 

Shouldn't we instead update the `getAllEbsVolumes` to filter out volumes that are attached to instances? In general, `aws-nuke` should have terminated all instances *before* deleting volumes, so the only volumes still attached are probably those that we aren't supposed to delete anyway...

Reply:
The would be a good approach if `getAllEbsVolumes` wasn't called _before_ any resources are deleted. The `getAll*` functions are used to populate the list the user sees before they confirm a nuke. The result of those functions are also reused for nuking so we don't have to make a duplicate api call. One way I could go is use check if the volume has any `VolumeAttachments` and going ahead to delete if it doesn't

Reply:
Ah, gotcha. Yea, I think just before deleting each volume, you'll have to check if it's attached, and skip it (while logging a warning) if it is.

Reply:
I'm curious about something. `VolumeInUse` is thrown when a volume is still in use (whether or not it's a protected EC2 instance using it). If we're only checking for EC2 instances it means we could miss other scenarios that prevents a volume from being deleted which will again cause aws-nuke to fail. Granted my error message was a bit too specific, but my thinking is that it's better to catch all cases when it might be in use and update the log message to something a bit more generic e.g. "EBS volume vol-061bc7f7561ab1934 is attached to an active resource"

Reply:
That's a fair point. I think logging a warning with that more generic error message is good enough!

Reply:
Log message has been updated